docs/modules/ROOT/pages/explanations/decisions/capacity-alerting.adoc

```diff
@@ -0,0 +1,111 @@
+= Handling of Capacity Alerts for Incident Prevention
+
+== Problem
+
+All Databases by VSHN have limited capacity such as storage and memory capacity, or a limit on concurrent connections.
+Depending on the database type running out of capacity can lead to downtime or even to data loss.
+We need a way to predict capacity issues, so we can react and prevent an incident.
+
+Writing these predictive capacity alerts is generally a solved problem.
+We can write Prometheus queries that predict the resource usage and alert based on the predicted usage.
+The main question is how we want to handle these alerts.
+
+From experience from similar system, we know that in nearly all cases capacity issues can only be resolved by the service users themselves, either by freeing up resources or by scaling up the instance.
+So first and foremost we need a process to inform the service user about potential capacity issues.
+
+== Proposals
+
+=== Alert Responsible Ops
+
+The simplest approach would be to alert the engineer that does https://handbook.vshn.ch/role_responsibleops.html[responsible ops^].
+This is how we are used to handling alerts and how we handled this in previous systems.
+
+In practice this means we will deploy fixed Prometheus alert rules and alert-manager configs in every database namespace that will send alerts to Opsgenie.
```

Comment:
I don't understand the Alertmanager configs part.
Why would we need separate Alertmanager configs for every namespace to alert our ResponsibleOps?

Reply:
Because on APPUiO Cloud to send alerts to our ResponsibleOps we'll most likely need to use the user-workload monitoring for our alerts. That means we'll need to add Alertmanager config in every namespace we manage and can't depend on cluster-monitoring.